CMake/utils/kwiver-utils-msvc.cmake

```diff
@@ -0,0 +1,109 @@
+    file(READ "${setup_batch}" contents)
+    # Convert file contents into a CMake list
+    # Where each element in the list is one line of the file
+    string(REGEX REPLACE ";" "\\\\;" contents "${contents}")
+    string(REGEX REPLACE "\n" ";" contents "${contents}")
+    foreach(line ${contents})
+      string(STRIP "${line}" line)
+      string(TOLOWER "${line}" search)
+      # Ignore comments
+      string(FIND "${search}" "::" idx)
+      if("${idx}" GREATER -1)
+        continue()
+      endif()
+      string(FIND "${search}" "rem" idx)
+      if("${idx}" GREATER -1)
+        continue()
+      endif()
+      string(FIND "${search}" "set config" idx)
+      if("${idx}" GREATER -1)
+        continue()
+      endif()
+      # check for set
+      string(FIND "${search}" "set " idx)
+      string(FIND "${search}" "=" edx)
+      if(NOT "${idx}" EQUAL 0 OR ${edx} EQUAL -1)
+        continue()
+      endif()
+      
+      #message(STATUS "I cleaning ${line}")
+      string(LENGTH "${line}" length)
+      math(EXPR _vardx "${edx}-4")
+      math(EXPR _valdx "${edx}+1")
+      string(SUBSTRING "${line}" 4 ${_vardx} _var) # what are we setting?
+      string(SUBSTRING "${line}" ${_valdx} ${length} _val) # what are we setting it to?
+      # replace batch terms with msvc terms
+      string(REPLACE "%config%" "$(Configuration)" _val "${_val}")
+      string(REPLACE "%~dp0" "${batch_dir}" _val "${_val}")
+      # remove any recursive sets
+      string(REPLACE "%${_var}%" "" _val "${_val}")
+      #message(STATUS "I am setting ${_var} to ${_val}")
+      # Keep track of setting the same variable over and over
+      if(NOT ${_var} IN_LIST _env_variables)
+        list(APPEND _env_variables "${_var}")
+      endif()
+      list(APPEND "_env_${_var}" "${_val}")
+    endforeach()
+  endforeach()
+  # Consolidate setting env variables into one line    
+  foreach(_env_var ${_env_variables})
+    #message(STATUS "processing ${_env_var}")
+    set(MSVC_ENV "${MSVC_ENV}${_env_var}=")
+    foreach(_env_val ${_env_${_env_var}})
+      set(MSVC_ENV "${MSVC_ENV}${_env_val};")
+    endforeach()
+    set(MSVC_ENV "${MSVC_ENV}\n")
+  endforeach()
+  #message(STATUS "Setting MSVC environment to ${MSVC_ENV}")
+    
+  get_property(executables GLOBAL PROPERTY kwiver_executables)
+  get_property(executables_paths GLOBAL PROPERTY kwiver_executables_paths)
+  #message(STATUS "Create msvc files for these executables : ${executables}")
+  list(LENGTH executables stop)
+  math(EXPR stop "${stop}-1")
+  foreach(i RANGE 0 ${stop})
+    # split the exe into name and binary dir
+    list(GET executables ${i} exe)
+    list(GET executables_paths ${i} exe_binary_dir)
+    
+    #message(STATUS "Setting MSVC environment for ${exe} in ${exe_binary_dir}")
+    configure_file(
+      ${KWIVER_CMAKE_DIR}/vcxproj.user.in
+      ${exe_binary_dir}/${exe}.vcxproj.user
+      @ONLY
+    )
+  endforeach()
+  
+endfunction()
```

Comment:
I wasn't sure, from previous discussions, if this parsing of batch files is still needed with the addition of collecting a list of executables?

Reply:
Yes, I always need to parse the batch file, its was just a matter or when can I parse the batch file (i.e. when is it complete)

Since its not assumed to be complete until all program sub-directories are processed, as they could add something to the batch file, I had to keep a list of executable locations so that I can add a vcxproj file to those locations when the batch is processed.

Reply:
Just checking my faulty memory.